hooks/transactions/useBtcRbfTransactionData.ts

```diff
@@ -0,0 +1,48 @@
+import { useQuery } from '@tanstack/react-query';
+import { rbf } from '../../transactions';
+import { Account, BtcTransactionData, SettingsNetwork } from '../../types';
+import { BitcoinEsploraApiProvider, mempoolApi } from '../../api';
+import { RbfData, sortFees } from './helpers';
+
+type Props = {
+  account: Account | null;
+  transaction?: BtcTransactionData;
+  btcNetwork: SettingsNetwork;
+  esploraProvider: BitcoinEsploraApiProvider;
+  isLedgerAccount: boolean;
+};
+
+const useBtcRbfTransactionData = ({ account, transaction, btcNetwork, esploraProvider, isLedgerAccount }: Props) => {
+  const fetchRbfData = async (): Promise<RbfData | undefined> => {
+    if (!account || !transaction) {
+      return;
+    }
+
+    const rbfTx = new rbf.RbfTransaction(transaction, {
+      ...account,
+      accountType: account.accountType || 'software',
+      accountId: isLedgerAccount && account.deviceAccountIndex ? account.deviceAccountIndex : account.id,
+      network: btcNetwork.type,
+      esploraProvider,
+    });
+
+    const mempoolFees = await mempoolApi.getRecommendedFees(btcNetwork.type);
+    const rbfRecommendedFees = await rbfTx.getRbfRecommendedFees(mempoolFees);
+    const rbfTransactionSummary = await rbf.getRbfTransactionSummary(esploraProvider, transaction.txid);
+
+    return {
+      rbfTransaction: rbfTx,
+      rbfTxSummary: rbfTransactionSummary,
+      rbfRecommendedFees: sortFees(rbfRecommendedFees),
+      mempoolFees,
+    };
+  };
+
+  return useQuery({
```

Comment:
i don't think we should add RQ in core, basically cause add another layer of cache that we need to worry about. but if we do, can we have query config object as an optional param? that way we can customize it in case of need. wdyt? @dhriaznov @teebszet

Reply:
@fedeerbes we discussed this question on the daily today, the decision for now is to try to add it since we'll need the same logic on both mobile and web extension, but we're going to be cautious with it and figuring out whether it adds any issues.